src/source/ContinuousRetryStreamCallbacks.h

```diff
@@ -33,9 +33,22 @@ struct __StreamLatencyStateMachine;
 struct __ConnectionStaleStateMachine;
 
 typedef struct __CallbackStateMachine {
+    // Indicator of stream being ready
+    volatile BOOL streamReady;
```

Comment:
Will any of these be read/written by different threads?  If so we should make them ATOMIC_BOOL

Reply:
These will get flagged by TSAN at some stage. I am thinking of bulk resolving TSAN issues at once. There are many false-positives that TSAN reports - for instance, these will be reported (if they are) but there is no chance of a race as I believe these are lock protected.

At any stage, we will do TSAN passes once we are done with the current issues

Reply:
Is there a reason to not opt for ATOMIC_BOOL by default just due to the nature of there being multiple threads here?  Unless of course if the atomic unit that is needed for correctness is beyond one variable then I agree we should prefer locking.

Reply:
I do think though we should investigate each TSAN issue one at a time and not bulk resolve, it's possible some of them are actually severe.

Reply:
Bulk-resolve doesn't mean shutting TSAN with suppressions. I meant go over the codebase and understand every reported instance. We do have a very high noise ratio with the reported issues. For example, in WebRTC, we have up-to 80% of the reported issues being benign but we still provided sometimes quirky fixes to get to as clean of a TSAN run as possible.

For this commit though, I don't want to extend to TSAN false-positives.

Reply:
OK yeah it looks like it's always being modified under the syncLock